cupyx/scipy/sparse/csr.py

```diff
@@ -215,13 +215,38 @@ def eliminate_zeros(self):
         self.indices = compress.indices
         self.indptr = compress.indptr
 
-    def maximum(self, other):
-        # TODO(unno): Implement maximum
+    def _maximum_minimum(self, other, cupy_op, op_name, dense_check):
+        if util.isscalarlike(other):
+            other = cupy.asarray(other, dtype=self.dtype)
+            if dense_check(other):
+                # Note: This is a work-around to make the output dtype the same
+                # as SciPy. It might be SciPy version dependent.
+                dtype = numpy.promote_types(self.dtype, numpy.float64)
```

Comment:
I think that aligning this check with the ones in https://github.com/cupy/cupy/pull/3905
is better and safer.